tests/lib/jira_api/test_jira_issue.py

```diff
@@ -11,250 +11,141 @@
-
-        # Assertions
-        assert task_id == "123"
-        mock_conn.create_issue.assert_called_once()
+@pytest.mark.parametrize("epic_id, should_call_epic", [(None, False), ("EPIC-1", True)])
+def test_create_jira_task(
+    mock_jira, mock_account, mock_issue_details, epic_id, should_call_epic
+):
+    """Tests task creation with and without an epic"""
+    mock_issue_details.epic_id = epic_id
+    mock_jira.create_issue.return_value = MagicMock(id="123")
 
+    task_id = create_jira_task(mock_account, mock_issue_details)
 
-def test_create_jira_task_with_epic_success():
-    """
-    Tests a task is created when providing an epic
-    """
-    with patch("jira.client.JIRA") as mock_conn:
-        mock_account = MagicMock()
-
-        mock_conn = mock_conn.return_value
-        mock_conn.create_issue.return_value = MagicMock(id="123")
-
-        # Create a mock IssueDetails object
-        issue_details = MagicMock()
-
-        # Call the function
-        task_id = create_jira_task(mock_account, issue_details)
-
-        # Assertions
-        assert task_id == "123"
-        mock_conn.create_issue.assert_called_once()
-        mock_conn.add_issues_to_epic.assert_called_once()
-
-
-def test_add_comment_calls_jira():
-    """
-    Tests that add_comment correctly calls the JIRA API
-    """
-    with patch("jira.client.JIRA") as mock_conn:
-        mock_account = MagicMock()
-
-        mock_conn = mock_conn.return_value
-
-        # Call the function
-        add_comment(mock_account, "ISSUE-123", "This is a test comment", internal=True)
-
-        # Assertions
-        mock_conn.add_comment.assert_called_once_with(
-            "ISSUE-123", "This is a test comment", True
-        )
-
-
-def test_add_comment_with_internal_false():
-    """
-    Tests that add_comment correctly calls the JIRA API with internal=False
-    """
-    with patch("jira.client.JIRA") as mock_conn:
-        mock_account = MagicMock()
-
-        mock_conn = mock_conn.return_value
-
-        # Call the function
-        add_comment(mock_account, "ISSUE-123", "Public comment", internal=False)
+    assert task_id == "123"
+    mock_jira.create_issue.assert_called_once()
+    if should_call_epic:
+        mock_jira.add_issues_to_epic.assert_called_once()
+    else:
+        mock_jira.add_issues_to_epic.assert_not_called()
 
-        # Assertions
-        mock_conn.add_comment.assert_called_once_with(
-            "ISSUE-123", "Public comment", False
-        )
 
+@pytest.mark.parametrize(
+    "comment_text, internal_flag",
+    [("This is a test comment", True), ("Public comment", False)],
+)
+def test_add_comment(mock_jira, mock_account, comment_text, internal_flag):
+    """Tests add_comment with different internal flag values"""
+    add_comment(mock_account, "ISSUE-123", comment_text, internal=internal_flag)
+    mock_jira.add_comment.assert_called_once_with(
+        "ISSUE-123", comment_text, internal_flag
+    )
```

Comment:
I think renaming the variables this way is clearer.
`comment_text -> comment` and `internal_flag -> is_internal`
That way:
comment = "This is a comment"
is_internal = True

```suggestion
@pytest.mark.parametrize(
    "comment, is_internal",
    [("This is a test comment", True), ("Public comment", False)],
)
def test_add_comment(mock_jira, mock_account, comment, is_internal):
    """Tests add_comment with normal and internal comments."""
    add_comment(mock_account, "ISSUE-123", comment, internal=is_internal)
    mock_jira.add_comment.assert_called_once_with(
        "ISSUE-123", comment, is_internal
    )
```

Reply:
fair enough. Changed in the code.